google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/it/ITBigQueryTimeEncoderTest.java

```diff
@@ -0,0 +1,158 @@
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "e32_time_seconds", StandardSQLTypeName.INT64)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "e64_time_micros", StandardSQLTypeName.INT64)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "e64_time_nanos", StandardSQLTypeName.INT64)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "e64_datetime_seconds", StandardSQLTypeName.INT64)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "e64_datetime_micros", StandardSQLTypeName.INT64)
+                        .build()
+                )))
+            .build();
+    bigquery.create(tableInfo);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (client != null) {
+      client.close();
+    }
+    if (bigquery != null) {
+      RemoteBigQueryHelper.forceDelete(bigquery, DATASET);
+    }
+  }
+
+  @Test
+  public void TestTimeEncoding() throws IOException, InterruptedException, ExecutionException,
+      Descriptors.DescriptorValidationException{
+    TableName parent = TableName.of(ServiceOptions.getDefaultProjectId(), DATASET, TABLE);
+    try (JsonStreamWriter jsonStreamWriter =
+        JsonStreamWriter.newBuilder(parent.toString(), tableInfo.getDefinition().getSchema())
+            .createDefaultStream()
+            .build()) {
+      JSONObject row = new JSONObject();
+      row.put("test_str", "Start of the day");
+      row.put("e32_time_seconds", CivilTimeEncoder.encodePacked32TimeSeconds(LocalTime.of(1,1,0)));
+      row.put("e64_time_micros", CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(1,1,0)));
+      row.put("e64_time_nanos", CivilTimeEncoder.encodePacked64TimeNanos(LocalTime.of(1,1,0)));
+      row.put("e64_datetime_seconds", CivilTimeEncoder.encodePacked64DatetimeSeconds(
+          LocalDateTime.of(1,1,1,1,1,1)));
+      row.put("e64_datetime_micros", CivilTimeEncoder.encodePacked64DatetimeMicros(
+          LocalDateTime.of(1,1,1,1,1, 1,1_000_000)));
+      JSONArray jsonArr = new JSONArray(new JSONObject[] {row});
+      ApiFuture<AppendRowsResponse> response = jsonStreamWriter.append(jsonArr, -1);
+      Assert.assertFalse(response.get().getAppendResult().hasOffset());
+      TableResult result =
+          bigquery.listTableData(
+              tableInfo.getTableId(), BigQuery.TableDataListOption.startIndex(0L));
+      Iterator<FieldValueList> iter = result.getValues().iterator();
+      FieldValueList currentRow;
+      currentRow = iter.next();
+      assertEquals("Start of the day", currentRow.get(0).getValue());
+      assertEquals(LocalTime.of(1,1,0),
+          CivilTimeEncoder.decodePacked32TimeSeconds(
```

Comment:
The returned field is string, and how could it be decoded like this?

Reply:
It actually returns at Java.time.LocalTime, I may have changed it after writing this test though.

Reply:
Now that it's private I will remove this and I don't think we need to worry about it